mlflow/server/handlers.py

```diff
@@ -293,10 +292,12 @@ def _search_runs():
     run_view_type = ViewType.ACTIVE_ONLY
     if request_message.HasField('run_view_type'):
         run_view_type = ViewType.from_proto(request_message.run_view_type)
-    sf = SearchFilter(filter_string=request_message.filter)
+    filter_string = request_message.filter
```

Comment:
Refactored `SearchFilter` to remove the statefulness -- we now just pass around strings everywhere and do the parsing at the last possible moment using `SearchUtils`.

Reply:
This is a great refactor!